### PR TITLE
provide detailed information on recovering from version mismatch

### PR DESCRIPTION
**Pull request by TomasTomecek**
Related: #873

### CODE REVIEW

**Comment by TomasTomecek** [comment]
rebased

**Comment by TomasTomecek** [comment]
rebased; ping @dnephin @shin-

File: docker/client.py [hunk 1]
```diff
@@ -151,6 +151,18 @@ def _raise_for_status(self, response, explanation=None):
         except requests.exceptions.HTTPError as e:
             if e.response.status_code == 404:
                 raise errors.NotFound(e, response, explanation=explanation)
+            if e.response.status_code == 400:
+                if response.content.startswith("client is newer than server"):
+                    explanation = (
+                        'docker-py is using newer API version than docker'
```
Comment (dnephin) [inline docker/client.py]: It seems a bit strange to me that a library would report itself in the error message.

Usually a library is consumed by some other application, which probably doesn't want to expose these details in error messages.

I don't really know how I feel about this change. I think a better solution might be to just include the error string in the docs, that way if someone hits the issue and they search for it, they'll find the docs about how to set the version.

**Comment by dnephin** [comment]
I think something like this is probably a good idea. I'm not sure if "Pitfalls" is the right term. Maybe "Gotchas" or just FAQ?

**Comment by TomasTomecek** [comment]
since I'm not native speaker, I usually have hard times picking the right term; I really like FAQ, will change it to that one, thanks for suggestion

**Comment by TomasTomecek** [comment]
@dnephin updated

**Comment by dnephin** [comment]
LGTM

**Comment by TomasTomecek** [comment]
CI failure seems unrelated:

```
docker: Error response from daemon: Cannot link to a non running container: /dpy-dind-597 AS /dpy-dind2-597/docker.
docker: Error response from daemon: Cannot link to a non running container: /dpy-dind-597 AS /dpy-dind3-597/docker.
```

**Comment by TomasTomecek** [comment]
rebased

**Comment by TomasTomecek** [comment]
rebased

**Comment by TomasTomecek** [comment]
rebased

@shin- can we merge this, it already has one lgtm from @dnephin

**Comment by bfirsh** [comment]
Shouldn't this be in the part of the documentation about instantiating clients? Creating an FAQ is always a warning sign for me that the main documentation isn't good enough.

**Comment by TomasTomecek** [comment]
updated, @bfirsh PTAL

**Comment by TomasTomecek** [comment]
@shin- @bfirsh rebased

**Comment by TomasTomecek** [comment]
rebased

**Comment by TomasTomecek** [comment]
\o/ thank you!

**Pull request merged**